watch.cpp

```diff
@@ -0,0 +1,135 @@
+#include "watch.hpp"
+
+#include "common_types.hpp"
+
+#include <fmt/format.h>
+
+#include <phosphor-logging/elog-errors.hpp>
+
+namespace hw_isolation
+{
+namespace watch
+{
+namespace inotify
+{
+
+using namespace std::string_literals;
+using namespace phosphor::logging;
+
+Watch::Watch(const sd_event* eventObj, const int inotifyFlagsToWatch,
+             const uint32_t eventMasksToWatch, const uint32_t eventsToWatch,
+             const std::filesystem::path& fileToWatch,
+             WatcherHandler watcherHandler) :
+    _inotifyFlagsToWatch(inotifyFlagsToWatch),
+    _eventMasksToWatch(eventMasksToWatch), _eventsToWatch(eventsToWatch),
+    _fileToWatch(fileToWatch), _watcherHandler(watcherHandler),
+    _watchDescriptor(-1), _watchFileDescriptor(inotifyInit())
+{
+    if (!std::filesystem::exists(_fileToWatch))
```

Comment:
added the points in previous pathc related ti filesystem existence in genesis boot